perses/app/relative_setup.py

```diff
@@ -159,10 +160,46 @@ def __init__(self,
         self._solvent_box_dimensions = solvent_box_dimensions
         self._use_given_geometries = use_given_geometries
         self._given_geometries_tolerance = given_geometries_tolerance
-        self._ligand_input = ligand_input
+        self.small_molecule_forcefield = small_molecule_forcefield
+        if ligand_input:
+            self._ligand_input = AnyPath(ligand_input)
+
+        if protein_pdb_filename:
+            self.protein_pdb_filename = AnyPath(protein_pdb_filename)
+        else:
+            self.protein_pdb_filename = protein_pdb_filename
+
+        if receptor_mol2_filename:
+            self.receptor_mol2_filename = AnyPath(receptor_mol2_filename)
+        else:
+            self.receptor_mol2_filename = receptor_mol2_filename
+
+        if small_molecule_parameters_cache:
+            self.small_molecule_parameters_cache = AnyPath(small_molecule_parameters_cache)
+        else:
+            self.small_molecule_parameters_cache = small_molecule_parameters_cache
+
+        if trajectory_prefix:
+            self._trajectory_prefix = AnyPath(trajectory_prefix)
+        else:
+            self._trajectory_prefix = trajectory_prefix
+
+        if trajectory_directory:
+            self._trajectory_directory = AnyPath(trajectory_directory)
+        else:
+            self._trajectory_directory = trajectory_directory
+
+        if forcefield_files:
+            if isinstance(forcefield_files, (list, set, tuple)):
+                self.forcefield_files = [AnyPath(forcefield_file) for forcefield_file in forcefield_files]
+            else:
+                self.forcefield_files = AnyPath(forcefield_files)
+        else:
+            self.forcefield_files = forcefield_files
```

Comment:
Am I correct in thinking that all these conditionals are because we want handle the cases where the given parameters are `None`? I know this doesn't look great, but I wonder if there is a better pattern to handle this, thoughts?

Reply:
Also, are we sure we want these to be public attributes that user can change? I know it might not be a big deal but I'm commonly in favor of trying to expose less, as possible, for maintainability purposes and avoiding future API breaking changes.

Reply:
Yes I hate it as well. It could be condensed into a line using a ternary expression, but idk if that is more readable:
```python
self._trajectory_prefix = AnyPath(trajectory_prefix) if trajectory_prefix else trajectory_prefix
```
ternaries make sense to me but I've got a math background so it feels natural but I'm not sure if it is Python. 

The root cause of the mess is that we have some arguments that can be `None`, which will fail to cast into a path, so I check if they are `None`. The other way to do this is with `try..except` which IMHO is the more python way to do it (EAFP is kinda the python way https://stackoverflow.com/questions/11360858/what-is-the-eafp-principle-in-python)

Reply:
ternary expressions should be fine, but I don't think this is a blocker, so feel free to change them if you want. I agree that having to check if attributes are `None` is not great and is the root of having to manually check with these conditionals.